gtda/base.py

```diff
@@ -139,7 +141,10 @@ def transform_plot(self, X, sample=0, **plot_params):
             Transformed one-sample slice from the input.
 
         """
-        Xt = self.transform(X[sample:sample+1])
+        Xt = self.transform(X[sample:sample + 1])
+        if "sample_orig" not in plot_params:
+            if "sample_orig" in signature(self.plot).parameters:
+                plot_params["sample_orig"] = sample
```

Comment:
I do not understand why we need to verify `"sample_orig" in signature(self.plot).parameters`. Would it be erroneous to put `sample_orig` in all parameters?

Reply:
It wouldn't be wrong, but what is the need for doing it? Not sure consistency for its own sake is worth it when introducing unused parameters, but maybe I'm missing something.

Reply:
The implicit point is that we do not have titles in all `plot` methods and this PR did not originally seek to change that. It could, of course.

Reply:
Ah, I see, my bad :facepalm: .